azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/util/AzureTokenCredentialsConverter.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.lib.auth.util;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.management.AzureEnvironment;
+import com.microsoft.azure.credentials.AzureTokenCredentials;
+import com.microsoft.azure.toolkit.lib.auth.AzureTokenCredentialsAdapter;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Arrays;
+
+public class AzureTokenCredentialsConverter {
+    public static AzureTokenCredentials convert(AzureEnvironment env, String tenantId, TokenCredential tokenCredential) {
+        return new AzureTokenCredentialsAdapter(toV1Environment(env), tenantId, tokenCredential);
+    }
+
+    private static com.microsoft.azure.AzureEnvironment toV1Environment(AzureEnvironment env) {
```

Comment:
seems this method can be embedded inplace since it's only used by the upper method.